composer/lib/dependabot/composer/update_checker/version_resolver.rb

```diff
@@ -37,8 +37,9 @@ def initialize(extensions)
             (?<=require\s)php(?:\-[^\s\/]+)?\s.*?\s(?=->) # composer v2
           }x
         VERSION_REGEX = /[0-9]+(?:\.[A-Za-z0-9\-_]+)*/
-        SOURCE_TIMED_OUT_REGEX =
-          /The "(?<url>[^"]+packages\.json)".*timed out/
+
+        # Example Timeout error from Composer 2.7.7: "curl error 28 while downloading https://example.com:81/packages.json: Failed to connect to example.com port 81 after 9853 ms: Connection timed out" # rubocop:disable Layout/LineLength
+        SOURCE_TIMED_OUT_REGEX = %r{curl error 28 while downloading (?<url>https?://.+/packages\.json): }
```

Comment:
I checked multiple examples of `composer` timeout messages that I found through googling, and the common thing for all of them was `curl error 28` which is `curl`'s well-known error code for timeouts, so I went with regex'ing on that...

Reply:
Could we simplify to just `curl error 28`? the rest of the string seems like it's more likely to change by copy updates and might not be necessary here?

Reply:
I had the same thought originally 😄 , but the regex capture string is also used in the code logic + error message:
https://github.com/dependabot/dependabot-core/blob/725e5adb2967f861a03bc568d467bd0034fa9138/composer/lib/dependabot/composer/update_checker/version_resolver.rb#L329-L337

We could monkey with it by replacing `while downloading` with `.+`, but that feels risky to me... for example, if there's second URL it could grab the wrong URL and end up causing more confusion like what happened in:
* https://github.com/dependabot/dependabot-core/issues/4625

Reply:
We could split looking for the error code as a hard coded string and then do the regex to find the URL... 

But I'm not sure it's worth the hassle now that we have a passing unit test which should catch regressions?

Currently this is broken on `main` and has been for several years with no complaints so this isn't worth spending a lot of time on. I'm only spending time fixing it now because I had this open PR for several years waiting til we dropped `composer v1`.

Reply:
Yeah I' with you 👍 thanks for elaborating